apps/remote_control/test/lexical/remote_control/code_intelligence/rename_test.exs

```diff
@@ -0,0 +1,338 @@
+      {:ok, result} =
+        ~q[
+        defmodule Baz do
+          alias |Foo
+        end
+      ] |> rename("Renamed")
+
+      assert result =~ ~S[alias Renamed]
+    end
+
+    test "succeeds when the definition in a nested module" do
+      {:ok, result} =
+        ~q[
+        defmodule TopLevel do
+          defmodule Foo do
+          end
+        end
+
+        defmodule TopLevelTest do
+          alias TopLevel.|Foo
+        end
+      ] |> rename("Renamed")
+
+      assert result == ~q[
+        defmodule TopLevel do
+          defmodule Renamed do
+          end
+        end
+
+        defmodule TopLevelTest do
+          alias TopLevel.Renamed
+        end
+      ]
+    end
+
+    test "succeeds when the cursor in the multiple aliases off of single alias" do
+      {:ok, result} =
+        ~q[
+        defmodule TopLevel do
+          alias Foo.{
+            First, Second,
+            Third.Fourth
+          }
+          |Second
+        end
+      ] |> rename("Renamed")
+
+      assert result =~ ~S[  First, Renamed,]
+    end
+
+    test "only rename the aliased when the cursor at the aliased" do
+      {:ok, result} =
+        ~q[
+        defmodule TopLevel do
+          alias Foo.Bar, as: FooBar
+          |FooBar
+        end
+        ]
+        |> rename("Renamed")
+
+      assert result =~ ~S[alias Foo.Bar, as: Renamed]
+      assert result =~ ~S[  Renamed]
+    end
+
+    test "succeeds when the cursor at the alias_ased child" do
+      {:ok, result} =
+        ~q[
+          defmodule TopLevel.Foo.Bar do
+          end
+
+          defmodule TopLevel.Another do
+            alias TopLevel.Foo, as: Parent
+            Parent.|Bar
+          end
+        ]
+        |> rename("Renamed")
+
+      assert result =~ ~S[defmodule TopLevel.Foo.Renamed]
+
+      assert result =~ ~S[  Parent.Renamed]
+    end
+
+    test "only rename aliased when the cursor at the alias_ased" do
+      {:ok, result} =
+        ~q[
+          defmodule TopLevel.Foo.Bar do
+          end
+
+          defmodule TopLevel.Another do
+            alias TopLevel.Foo, as: Parent
+            |Parent.Bar
+          end
+        ]
+        |> rename("Renamed")
+
+      assert result =~ ~S[defmodule TopLevel.Foo.Bar do]
+
+      assert result =~ ~S[alias TopLevel.Foo, as: Renamed]
+      assert result =~ ~S[  Renamed.Bar]
+    end
+
+    test "shouldn't rename the relative module" do
+      {:ok, result} =
+        ~q[
+        defmodule |Foo do
+        end
+
+        defmodule FooTest do
+        end
+        ] |> rename("Renamed")
+
+      assert result =~ ~S[defmodule FooTest do]
+    end
+
+    test "shouldn't rename the descendants when the cursor at the end of the module" do
+      {:ok, result} = ~q[
+        defmodule TopLevel.Module do # ✓
+        end
+
+        defmodule TopLevel.Module.Another do # x
+          alias TopLevel.|Module
+        end
+      ] |> rename("Renamed")
+
+      refute result =~ ~S[defmodule TopLevel.Renamed.Another]
```

Comment:
i would definitely expect the module Toplevel.Module.Another to be renamed to `Toplevel.Renamed.Another`

Reply:
Here, I would like to preserve the functionality of renaming an exact module.

Consider the following scenario: Let's say we have two modules, `Project.Schema.User` and `Project.Schema`. If we find that `Project.Schema` is not clear in its meaning and rename it to `Project.EctoHelper`, should we also rename `Project.Schema.User` to `Project.EctoHelper.User`?

I believe the answer is no. Clearly, there are times when we want to rename a specific module, and there are times when we want to rename all modules that have that module as a prefix.